caiman/base/timeseries.py

```diff
@@ -168,27 +172,43 @@ def save(self, file_name, to32=True, order='F'):
         name, extension = os.path.splitext(file_name)[:2]
         print(extension)
 
+        if to32 and not('float32' in str(self.dtype)):
+            mn,mx = np.percentile(self,[.01,99.999])
+            
         if extension == '.tif':  # load avi file
-            try:
-
-                from tifffile import imsave
-                print('tifffile package not found, using skimage instead for imsave')
-
-            except:
 
-                from skimage.external.tifffile import imsave
-            if to32:
-                np.clip(self, np.percentile(self, 1),
-                        np.percentile(self, 99.99999), self)
-                minn, maxx = np.min(self), np.max(self)
-                data = 65536 * (self - minn) / (maxx - minn)
-                data = data.astype(np.int32)
-                imsave(file_name, self.astype(np.float32))
-            else:
-                imsave(file_name, self)
+                
+            with tifffile.TiffWriter(file_name, bigtiff=bigtiff, imagej=imagej, software=software) as tif:
+                
+                    
+                for i in range(self.shape[0]):
+                    if i%200 == 0:
+                        print(str(i) + ' frames saved')
+                    
+                    curfr = self[i].copy()
+                    if to32 and not('float32' in str(self.dtype)):
+                         curfr = np.clip(curfr, mn, mx)                
+                         curfr = 65536 * (curfr - mn) / (mx - mn)
```

Comment:
where does this come from?

Reply:
float32 ranges in [-2^15 + ,2^15-1] which is 32767. But most importantly why do you multiply with this? This would explain some weird behavior I've seen

Reply:
You are right. It was there before and I did not correct it, thank you for spotting it. I assume in the past the assumption was that it was always positive. Now we now it is not the case.

On the other hand while I was writing the code I wondered whether this is necessary at all since this is floating point...

Reply:
This behavior in the past should not have affected much since we seldom save with this function

Reply:
this would happen if you would try to motion correct a uint16 image that was suddenly acquiring negative values. Now things are converted into single so it's shouldn't matter but let's look into it tomorrow.

Reply:
Wait, 65535 should be OK since I make it positive. Let's chat about this tomorrow.